contracts/SpokePool.sol

```diff
@@ -667,6 +674,58 @@ abstract contract SpokePool is
         _emitFillRelay(relayExecution, fillAmountPreFees);
     }
 
+    /**
+     * @notice Caller signals to the system that they want a refund on this chain, which they set as the
+     * `repaymentChainId` on the original fillRelay() call on the `destinationChainId`. An observer should be
+     * be able to 1-to-1 match the emitted RequestRefund event with the FilledRelay event on the `destinationChainId`.
+     * @dev This function could be used to artificially inflate the `fillCounter`, allowing the caller to "frontrun"
+     * and cancel pending fills in the mempool. This would worst case censor fills at the cost of the caller's
+     * gas costs. We don't view this as a major issue as the fill can be resubmitted and obtain the same incentive,
+     * since incentives are based on validated refunds and would ignore these censoring attempts. This is no
+     * different from calling `fillRelay` and setting msg.sender = recipient.
+     * @param relayer Relayer on FilledRelay event on destinationChain we want to match with this Refund.
+     * This should be account receiving any refund stemming from this event.
+     * @param destinationToken This chain's destination token equivalent for original deposit destination token.
+     * @param amount Original deposit amount.
+     * @param originChainId Original origin chain ID.
+     * @param realizedLpFeePct Original realized LP fee %.
+     * @param depositId Original deposit ID.
+     * @param maxCount Max count to protect the refund recipient from frontrunning.
+     */
+    function requestRefund(
+        address relayer,
+        address destinationToken,
+        uint256 amount,
+        uint256 originChainId,
+        int64 realizedLpFeePct,
+        uint32 depositId,
+        uint256 maxCount
+    ) external nonReentrant {
+        // TODO: We don't check for `amount > 0` but maybe we should? `fillRelay` reverts
```

Comment:
Nit: would it make sense to perform the check here as a sanity check?

Reply:
Yeah I added a check that `amount > 0`